workflow-templates/deploy-pages.yml

```diff
@@ -35,19 +35,23 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Setup Pages
-        uses: actions/configure-pages@v3
+        uses: actions/configure-pages@v5
 
-      - name: Build Documentation
-        uses: ammaraskar/sphinx-action@0.4
-        with:
-          pre-build-command: "pip install ${{ github.event.repository.name }}"
-          docs-folder: "docs/"
+      - name: Install the ${{ github.event.repository.name }} Package
+        run: |
+          cd ./python
+          pip install -e .[all]
+
+      - name: Build ${{ github.event.repository.name }} Documentation
+        run: |
+          cd ./docs
+          sphinx-build source doc_artifacts
 
-      - name: Upload Artifact
-        uses: actions/upload-pages-artifact@v2
+      - name: Upload ${{ github.event.repository.name }} Documentation Artifact
+        uses: actions/upload-pages-artifact@v3
         with:
-          path: ./docs/docs/html
+          path: ./docs/doc_artifacts
 
       - name: Deploy to GitHub Pages
         id: deploy
-        uses: actions/deploy-pages@v2
+        uses: actions/deploy-pages@v4
```

Comment:
Since added these two steps and bumped as well, can go ahead and link their associated issues to this PR.